src/slider/slider.js

```diff
@@ -0,0 +1,133 @@
+angular.module('ui.bootstrap.slider', ['ui.bootstrap.position'])
+
+.constant('sliderConfig', {
+  updateOnDrag: false,
+  max: 100,
+  min: 0
+})
+
+.directive('slider', ['$document', '$position', 'sliderConfig', function($document, $position, sliderConfig) {
+    return {
+      restrict: 'EA',
+      replace: true,
+      require: '?ngModel',
+      transclude: true,
+      scope: {
+        transition: '@',
+        barClass: '@',
+        btnClass: '@'
+      },
+      templateUrl: 'template/slider/slider.html',
+      link: function(scope, element, attrs, ngModel) {
+        scope.max = angular.isDefined(attrs.max) ? scope.$parent.$eval(attrs.max) : (sliderConfig.max || 100);
+        scope.min = angular.isDefined(attrs.min) ? scope.$parent.$eval(attrs.min) : (sliderConfig.min || 0);
+        scope.transition = scope.transition || sliderConfig['transition'];
+        scope.updateOnDrag = angular.isDefined(attrs.updateOnDrag) ? scope.$parent.$eval(attrs.updateOnDrag) : (sliderConfig['updateOnDrag'] || false);
+
+        scope.btn = element.find('a');
+        scope.bar = element.children().eq(1).children().eq(0);
+        scope.range = scope.max - scope.min;
+
+        scope.hasAnimation = (scope.transition) && (scope.transition !== 'none');
+
+        scope.setAnimation = function(doanimate) {
```

Comment:
I would recommend moving this function into a directive controller and requiring that controller.